packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -152,6 +151,13 @@ async function instantiateModule(
     throw e
   }
 
+  if (!ssrModule.__esModule) {
+    Object.defineProperty(ssrModule, '__esModule', { value: true })
+    if (!Object.getOwnPropertyDescriptor(ssrModule, 'default')) {
+      Object.defineProperty(ssrModule, 'default', { value: ssrModule })
```

Comment:
This makes a CommonJS module compatible with ESM default import.

ES modules are not affected, since `ssrTransform` now injects a `Object.defineProperty` call when ESM syntax is detected.